modules/programs/zellij.nix

```
@@ -0,0 +1,52 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.zellij;
+  yamlFormat = pkgs.formats.yaml { };
+in {
+  options = {
+    programs.zellij = {
+      enable = mkEnableOption "zellij";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.zellij;
+        defaultText = literalExpression "pkgs.zellij";
+        example = literalExpression "pkgs.zellij";
+        description = ''
+          The zellij package to install.
+        '';
+      };
+
+      settings = mkOption {
+        type = yamlFormat.type;
+        default = { };
+        defaultText = "{}";
+        example = literalExpression ''
+          {
+            theme = "custom";
+            themes.custom.fg = 5;
+          }
+        '';
+        description = "The zellij configuration.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ cfg.package ];
+    xdg.configFile."zellij/config.yaml" = mkIf (cfg.settings != { }) {
+      source = yamlFormat.generate "zellij.yaml" cfg.settings;
+    };
+  };
+
+  meta = {
+    description = "A terminal workspace with batteries included.";
+    homepage = "https://zellij.dev/";
+    maintainers = [ maintainers.mainrs ];
+    platforms = platforms.unix;
+    license = licenses.mit;
+  };
```

```suggestion
  meta.maintainers = [ maintainers.mainrs ];
```
AFAIK the module system does not use/have other meta attributes.

---

I'll change this in one of my other PRs for `eww` as well! Thanks for letting me know!